docs/DeveloperGuide.md

```diff
@@ -1,34 +1,178 @@
 # Developer Guide
 
-## Design & implementation
+![Supported Java versions](https://img.shields.io/badge/Java-11-blue.svg)
+![Supported OS](https://img.shields.io/badge/Supported%20OS-Windows|MacOS|Linux-yellow.svg)
+![Java CI](https://github.com/AY2021S1-CS2113T-T09-4/tp/workflows/Java%20CI/badge.svg)
+
+## Table of Contents
+
+- [Introduction](#introduction)
+    * [To be added](#to-be-added)
+        + [To be added](#to-be-added)
+
+## 1. Introduction
+
+### 1.1 Software Overview
+
+easyLog is a Command Line Interface (CLI) application for warehouse employees to manage items and orders in the
+warehouse. easyLog allows users to add new items and orders, view existing items and orders, remove specific items and
+orders and clear all existing items and orders in the warehouse at ease. If the user types fast, easyLog can get
+logistic management tasks done faster than traditional Graphical User Interface (GUI) applications.
+
+easyLog, comprises a Parser component, Ui component, Storage component, Item component, Order component and Command
+component. Each component consists of various classes that work in tandem, to ensure it meets the purpose of our
+program.
+
+The purpose of this developer guide is to allow any interested contributors, who wish to develop this application
+further or just curious about the workings of this application. This would allow potential contributors to be able to
+dive right in to improving the code, performance, features or even adding new features much more easily due to the
+understanding of the structure of the codebase and implementation of existing features.
+
+### 1.2 Prerequisites
+
+- Java 11 (can be downloaded from [here](https://www.oracle.com/sg/java/technologies/javase-jdk11-downloads.html))
+- IntelliJ Idea (can be downloaded from [here](https://www.jetbrains.com/idea/download/#section=mac))
+
+### 1.3 Setting up
+
+1. Fork the easyLog repository [here](https://github.com/AY2021S2-CS2113T-T09-4/tp) and git clone it to any location on
+   your computer.
+
+2. Open any IDE (IntelliJ Idea preferred as mentioned in the prerequisites) and click `Configure` -> `Project Defaults`
+   -> `Project Structure` -> `New`
+
+3. Next, go to `Import Project` and select the build.gradle file.
+
+4. After opening the project, go to `src` -> `main` -> `java` -> `seedu.easylog` -> EasyLog and right click on the
+   EasyLog class.
+
+5. Upon successful run, the following opening message will be shown:
+
+```
+Hello! I'm easyLog!
+What can I do for you? Enter help to view commands.
+Looking for save data.
+Save data not found.
+```
+
+## 2. Design
+
+### 2.1 Architecture
+
+![Architecture Diagram](https://user-images.githubusercontent.com/57165946/112990632-ddf7e780-9198-11eb-99ec-f29aafbfc04f.png)
```

Comment:
![image](https://user-images.githubusercontent.com/71376412/113077714-4203c580-9204-11eb-804b-e32d7ed138ca.png)

I'm not sure if this will count as a bug, but the aforementioned components are not mentioned in the document or the architecture diagram.